static/src/plugins/toaster.ts

```diff
@@ -0,0 +1,56 @@
+import { type App, h, render, type Plugin } from "vue";
+import type { ToasterApi } from "@/types";
+import type { ToastProps, ToastOptions } from "@/types";
+import ToastNotification from "../components/ToastNotification.vue";
+
+declare module "vue" {
+  interface ComponentCustomProperties {
+    $toast: ToasterApi;
+  }
+}
+
+function createToaster(app: App, globalOptions: ToastOptions): ToasterApi {
+  return {
+    show(message: string, options: ToastOptions = {}) {
+      const mergedOptions = { ...globalOptions, ...options };
+      const props: ToastProps = { message, ...mergedOptions };
+      const div = document.createElement("div");
+      document.body.appendChild(div);
+      const vNode = h(ToastNotification, props);
+      if (app && app._context) {
+        vNode.appContext = app._context;
+      }
+      render(vNode, div);
+      setTimeout(() => {
+        // The Toast just manages hiding itself but will get removed from the DOM here
+        if (div) {
+          render(null, div);
+          document.body.removeChild(div);
+        }
+      }, (options.duration?.valueOf() || 3000) + 1000);
```

Comment:
Why plus 1000? 
Also, I'd really prefer the default value closer to the ToastOptions, but I think I'm slowly getting a feeling for why this is not that simple...

Reply:
Could we maybe at least move such magic constants to named constants?